### PR TITLE
feat!: Introduced `--preset` and clean up config

### PR DESCRIPTION
**Pull request by fryorcraken**
# Description

**Breaking change**: TWN config is applied when `--preset=default` is passed instead of `--cluster-id=1`.

Clean-up config by reducing the number of arguments and also aligning default config with waku-org/nwaku-compose.

Note for reviewers: commits are kept clean, hence review per commit may be preferred.

# Changes

- [x] feat!: introduce `preset` option instead of deducing from cluster-id. `cluster-id=1` still applies TWN config for now.
- [x] chore: remove unused import warning
- [x] feat: set keep-alive to true for default preset (TWN)
- [x] feat: set max connections to 150 by default
- [x] feat!: change default eth client value to empty string
- [x] feat: remove relay peer exchange
- [x] feat: remove agentString option
- [x] chore: reorder maxMessageSize as it is not RLN specific
- [x] doc: clarify max shard value on CLI args
- [x] validate RLN Relay config

### CODE REVIEW

**Review by gabrielmer — COMMENTED** [review]
Thanks so much! I reaaaally like the changes!
The only thing I'm not sure about is removing Relay Peer Exchange.

I know Waku Peer Exchange and Discv5 are preferred, but it still works and is a valid discovery alternative

@jm-clius WDYT? Should it be removed?

File: waku/factory/waku.nim [hunk 1]
```diff
@@ -91,16 +91,8 @@ proc init*(T: type Waku, conf: WakuNodeConf): Result[Waku, string] =
...
-  # cluster-id=0
-  of 0:
-    let clusterZeroConf = ClusterConf.ClusterZeroConf()
-    confCopy.pubsubTopics = clusterZeroConf.pubsubTopics
-    # TODO: Write some template to "merge" the configs
-
-  # cluster-id=1 (aka The Waku Network)
-  of 1:
+  case confCopy.preset
+  of "default":
```
[the remark this answers is not in the document]
Comment (fryorcraken) [reply waku/factory/waku.nim:95]: The reasoning behind it is to start my moving away from "the waku network" terminology.

**Comment by fryorcraken** [comment]
Updates description with a todo list.

**Comment by SionoiS** [comment]
LGTM so far except, max connection is too low because the actual number of connection is divided by a factor for peer management purposes. egg. incoming relay peers use 17% of total, service peers 30%, etc...

**Comment by fryorcraken** [comment]
Best to merge https://github.com/waku-org/nwaku/pull/2779 first

**Comment by fryorcraken** [comment]
> LGTM so far except, max connection is too low because the actual number of connection is divided by a factor for peer management purposes. egg. incoming relay peers use 17% of total, service peers 30%, etc...
> 
> Having the actual numbers as config params could be done but is even more obscure params a good idea?

I think it's fine to keep a global parameter.

I used the default value from nwaku-compose. What value do you think should be set? I think we setup 300 in the fleets.

**Comment by SionoiS** [comment]
> I used the default value from nwaku-compose. What value do you think should be set? I think we setup 300 in the fleets.

200 is a bit low I feel. 300 gives us ~15 in and 20 out replay peers last I remember.

**Comment by fryorcraken** [comment]
Blocked by https://github.com/vacp2p/nim-libp2p/issues/1150 as I cannot compile nwaku anymore and hence cannot check failing tests locally.

**Comment by Ivansete-status** [comment]
@fryorcraken - is this still relevant or can be closed?

**Comment by fryorcraken** [comment]
Let's close for now.

This is relevant because as part of the _Messaging API_ I expect introduction of preset modes in both libwaku and wakunode2 which would impact behaviour at several level.
Eg edge vs relay vs service node
twn vs status 1:1
etc

**Pull request closed without merging**